### ISSUE TITLE
Translation

### ISSUE THREAD

**Issue by anmol098** [anmol098/waka-readme-stats#23]
Needed Translation of stats and hardcoded string to the language specified by the user.
and please contribute to the translation of readme also

**Comment by Tkd-Alex** [comment]
How I can help with in this issue?

**Comment by anmol098** [comment]
@Tkd-Alex you can find the `translation.json` file in the repo which can be used to translate the project to different languages. I have done for French same can be proceeded for different languages [final output can be found here](https://github.com/anmol098/anmol098/blob/master/Readme-fr.md) 

It will be really helpful if you can help to translate this project into a different locale.
Hoping for helpful contribution
Thanks

**Comment by hedythedev** [comment]
I can work on chinese

**Comment by guilyx** [comment]
I'll send some fixes for the french bits

**Comment by anmol098** [comment]
@hedythedev @Tkd-Alex thank you for translating to Chinese & Italian respectively

**Comment by anmol098** [comment]
@guilyx  contributions are most welcomed

**Comment by Na93r** [comment]
I can help with Arabic

**Comment by anmol098** [comment]
@Na93r Arabic translation is merged but for that, we have to make all stats to RTL as Arabic works in RTL. [check this out](https://github.com/anmol098/anmol098)

I think this has some issues in language reading?
correct me if I am wrong

**Comment by Na93r** [comment]
@anmol098 everything is correct except the shield.io
it cannot read "Profile Views" and "From Hello World I have written"
I will revert it back to English

**Comment by deborahbpc** [comment]
I could help with brazilian portuguese, if you need it

**Comment by dantehemerson** [comment]
I will work on es translations.

**Comment by tlatkdgus1** [comment]
I will work on korean translations.

but have some problem, The order of sentences in English and Korean is different.
Therefore, we are going to write transfer.json to enable the use of format string.

**Comment by iamgrodrigues** [comment]
I just add a translation to Brazilian Portuguese of your readme, and I also corrected some miss translation on the pt-BR section of the .json

**Comment by haydarKarkin** [comment]
I've opened a PR for Turkish translations. #160

**Comment by rust-am** [comment]
Hi! I can help with Russian.

**Comment by matiduda** [comment]
Happy to help with Polish :D

**Comment by AliAlmasi** [comment]
I can handle the Persian translation.

**Comment by infinotiver** [comment]
I can help with Hindi @anmol098 

👀  Can't find Hindi right now! How is that possible?

**Comment by whateverzpy** [comment]
Do you need the translation of the README file? I can help with Chinese translation.

**Comment by elouannh** [comment]
Hello, I contributed for the Japanese translation.

**Comment by mayannaoliveira** [comment]
Hello dear I can translate in Portuguese.

**Comment by SnowCutieOwO** [comment]
Done Simplified Chinese translation for README, along with some fixes for both Simplified Chinese and Traditional Chinese <3
#589

**Comment by mayannaoliveira** [comment]
I can help with Portuguese, please contact me in Discord (link in my bio).

**Comment by CleverLemming1337** [comment]
I would like to do the German translation.

**Comment by solusops** [comment]
Could we not use tools like DeepL or other Neural translation tools to translate on fly?
Instead of translating for each language, just brainstorming ideas.

**Comment by anmol098** [comment]
@DataBoySu That’s a good idea, and we can definitely move forward with it. If you could help us with a proof of concept (POC) and recommend an open-source AI model, that would be a valuable addition. Using AI, the metrics can be customized to a great extent, while also enhancing creativity.

**Comment by solusops** [comment]
@anmol098 sure, do you have a scope in mind....
as in what languages do you see being used in your repo?
and possible hosting/api thoughts if any

**Comment by anmol098** [comment]
@DataBoySu let's discuss this in detail feel free to book a meeting slot from here https://calendly.com/anmol098/30min

**Comment by guilyx** [comment]
> Could we not use tools like DeepL or other Neural translation tools to translate on fly? Instead of translating for each language, just brainstorming ideas.

What about cost ? Who pays the API calls ?

**Comment by anmol098** [comment]
> > Could we not use tools like DeepL or other Neural translation tools to translate on fly? Instead of translating for each language, just brainstorming ideas.
> 
> What about cost ? Who pays the API calls ?

There are lot of free SLM which can do the job better. I'm aware of together ai which provided deepseek and qwen for free. So each user can configure their together ai account and add the token. That should work.

**Comment by solusops** [comment]
> > > Could we not use tools like DeepL or other Neural translation tools to translate on fly? Instead of translating for each language, just brainstorming ideas.
> > 
> > 
> > What about cost ? Who pays the API calls ?
> 
> There are lot of free SLM which can do the job better. I'm aware of together ai which provided deepseek and qwen for free. So each user can configure their together ai account and add the token. That should work.

Yeah but, its not like we are going to need a lot of API calls, all we need is to translate the hard-coded english/stats/readme to various languages once and cache/store it

**Comment by solusops** [comment]
@anmol098 Hi anmol, I have been working on a translation pipeline for readme specifically
This was the result: https://github.com/DataBoySu/databoysu-readme-translator
I did not make it to completely replace humans but to get us 95% there, with minor edits and manual handling, it can generate great translations.
Here is the implementation I did on this repo's fork: https://github.com/DataBoySu/waka-readme-stats/tree/master
It needs some work but I think I will be done by tomorrow.

**Comment by solusops** [comment]
https://github.com/anmol098/waka-readme-stats/pull/606

PR made with the complete Readme pipeline with very necessary change to CI workflow to make translation.json updating more accessible to contributors.

**Comment by solusops** [comment]
> [!NOTE]
> translation.json has 21 languages now

> en: English
> fr: French
> bn: Bengali
> it: Italian
> zh: Chinese (Simplified)
> zh_TW: Chinese (Traditional)
> ar: Arabic
> pt: Portuguese(Brazilian)
> pl: Polish
> es: Spanish
> gl: Galician
> ko: Korean
> tr: Turkish
> ru: Russian
> de: German
> id: Indonesian
> ca: Catalan
> vn: Vietnamese
> uk: Ukrainian
> fa: Persian (Farsi)
> hi: Hindi

Readme now has a automated template translation built-in, #610 
Contributors can now edit/tweak and update it for more natural and native translation

> [!IMPORTANT]
> In case of future README overhaul, make sure to run the translation workflow responsibly, as it would overwrite potential contributor changes.

**Comment by lulushuz** [comment]
Hi @anmol098, I’d like to help with the Turkish localization. I noticed some parts in translation.json could be more natural, and I’ve already prepared a README.tr.md since it’s currently missing.

Should I go ahead and open a Pull Request for these?

**Comment by solusops** [comment]
> Hi [@anmol098](https://github.com/anmol098), I’d like to help with the Turkish localization. I noticed some parts in translation.json could be more natural, and I’ve already prepared a README.tr.md since it’s currently missing.
> 
> Should I go ahead and open a Pull Request for these?

Feel free to open a PR, and since you have already made a README.tr from scratch, its not a big deal however for README translations, you can simply fork the repo and run the workflow specifying the language code you want.
The purpose is to not have contributors work from scratch but to be given a automated language template to further improve upon.

**Comment by hobsonbobbi96-glitch** [comment]
not sure exactly how to do this can someone help me please?

**Comment by johnnylemonny** [comment]
Hi @anmol098! 👋

Just wanted to follow up here regarding the Polish localization. I've prepared a PR with updated translations (fixing typos, improving phrasing, and adding missing AI metrics) as well as the full Polish README (`README.pl.md`): #677.

Whenever you have a moment, I'd really appreciate it if you could take a look and review it. Thank you! 🙌
